fe/fe-core/src/main/java/com/starrocks/connector/ConnectorPartitionTraits.java

```diff
@@ -449,8 +450,59 @@ PartitionKey createEmptyKey() {
-            return null;
+            PaimonTable baseTable = (PaimonTable) table;
+            Set<String> result = Sets.newHashSet();
+            for (BaseTableInfo baseTableInfo : baseTables) {
+                if (!baseTableInfo.getTableIdentifier().equalsIgnoreCase(baseTable.getTableIdentifier())) {
+                    continue;
+                }
+                Optional<ConnectorMetadata> connectorMetadata = GlobalStateMgr.getCurrentState().getMetadataMgr().
+                        getOptionalMetadata(baseTable.getCatalogName());
+                if (!connectorMetadata.isPresent()) {
+                    LOG.error("Get paimon connectorMetadata failed : {}", baseTable.getCatalogName());
+                    throw new RuntimeException("Get paimon connectorMetadata failed :" + baseTable.getCatalogName());
+                }
+                Map<String, MaterializedView.BasePartitionInfo> partitionVersionMap =
+                        context.getBaseTableRefreshInfo(baseTableInfo);
+                List<String> partitions = getPartitionNames();
+                long mvLatestSnapShotID = Long.MIN_VALUE;
+                for (Map.Entry<String, MaterializedView.BasePartitionInfo> entry : partitionVersionMap.entrySet()) {
+                    if (entry.getValue() != null) {
+                        mvLatestSnapShotID = Math.max(mvLatestSnapShotID, entry.getValue().getVersion());
+                    }
+                    // If there are partitions deleted, return all latest partitions.
+                    if (!partitions.contains(entry.getKey())) {
+                        result.addAll(partitions);
+                        LOG.info("Get paimon updated partition names {}, partition {} has been deleted, return all.",
+                                baseTables, entry.getKey());
+                        return result;
+                    }
+                }
+                for (String part : partitions) {
+                    if (!partitionVersionMap.containsKey(part)) {
+                        result.add(part);
+                    }
+                }
+                ConnectorMetadata metadata = connectorMetadata.get();
+                List<PartitionInfo> changedPartitionInfo = metadata.getChangedPartitionInfo(baseTable,
+                        mvLatestSnapShotID);
+                for (PartitionInfo partitionInfo : changedPartitionInfo) {
+                    PaimonPartitionInfo info = (PaimonPartitionInfo) partitionInfo;
+                    // Change log record partition which has been deleted.
+                    if (!partitions.contains(info.getPartitionName())) {
+                        continue;
+                    }
+                    if (partitionVersionMap.containsKey(info.getPartitionName()) &&
+                            partitionVersionMap.get(info.getPartitionName()).getVersion() >= info.getModifiedTime()) {
+                        LOG.warn("Paimon changelog out of version, origin:{} , current:{}.",
```

Comment:
It's normal mv is later than the base table. no need to log a warning, it may be noisy if requests are frequent.

Reply:
partitionVersionMap store mv's partition version,  info store paimon's partition version. if partitionVersionMap.getVersion > info.version(), it means paimon has some problem such as roll back, which make mv's version more fresh than paimon. It's not nomal.